### PR TITLE
Nc games 6

### PR DESCRIPTION
**Pull request by sasjaWijdenes**

### CODE REVIEW

File: db/app.js [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
   app = express();
 
+app.use(express.json());
+
 app.get("/api/categories", getAllCategories);
 app.get("/api/users", getAllUsers);
 app.get("/api/reviews/:review_id", getReviewById);
 
+app.patch("/api/reviews/:review_id", incrementVotes);
+
```
Comment (two-triangles) [inline db/app.js:17]: This is a comment for line 25, I would move this to above your error handling as it isn't strictly an error handling function and is related to the endpoints

Comment (sasjaWijdenes) [reply db/app.js:17]: ah ok, after that am i good to merge?

File: db/controllers/reviews.controllers.js [hunk 2]
```diff
@@ -8,3 +11,20 @@ exports.getReviewById = (req, res, next) => {
...
     .catch(next);
 };
+
+exports.incrementVotes = (req, res, next) => {
+  const {
+    params: { review_id: id },
+    body: { inc_votes: votes },
+  } = req;
+  // console.log(typeof votes, ":", votes, "<<<votes");
+  if (!votes) next({ status: 404, msg: "Passed malformed body" });
```
Comment (two-triangles) [inline db/controllers/reviews.controllers.js:21]: I feel like the controller's job should be to deal with the request and response. It is the models job to handle the data that has been sent in with on the request and perform the requests to the database with that data. 

If there are no votes we would return a rejected Promise which would then be passed to the catch block of the controller, to then send on to the error handling middleware. 

I would move these errors to the model. Also regarding the checking of the typeof the number, this error will be handled by PSQL when you make the request.

File: db/models/reviews.models.js [hunk 3]
```diff
@@ -15,3 +15,14 @@ exports.fetchReviewById = (id) => {
       }
     });
 };
+exports.updateReviewVotes = (id, votesToAdd) => {
+  return this.fetchReviewById(id).then(({ votes }) => {
```
Comment (two-triangles) [inline db/models/reviews.models.js]: You don't need to make two requests to the database. You can perform the updating of the votes using only SQL. This was we don't have multple requests to a database for a request to a single endpoint.

**Comment by two-triangles** [comment]
Have a look over the comments and please make the amendments. I Will have a look over them before confirming merge

File: db/models/reviews.models.js (hunk 3 above)
Comment (sasjaWijdenes) [reply db/models/reviews.models.js]: @two-triangles I have made the suggested amends

File: db/controllers/reviews.controllers.js [hunk 4]
```diff
@@ -8,3 +11,20 @@ exports.getReviewById = (req, res, next) => {
...
+  if (typeof votes !== "number")
```
Comment (two-triangles) [inline db/controllers/reviews.controllers.js:22]: this should be in the model and you also have the error handler for checking if the votes exist in the model and controller. It should only be in the model

**Pull request merged**